gui/src-tauri/src/main.rs

```diff
@@ -148,12 +156,14 @@ fn main() {
 						Ok(()) => log::info!("send exit to backend"),
 						Err(_) => log::info!("fail to send exit to backend"),
 					}
-					// We can't wait the process to die because Tauri doesn't expose
-					// such thing, so the only way is sleeping and hope for the best
-					// for now...
-					// (We can wait pids in POSIX but not in Windows if we wanted
-					// to workaround this)
-					thread::sleep(Duration::from_secs(10))
+					let ten_seconds = Duration::from_secs(10);
+					let start_time = Instant::now();
+					while start_time.elapsed() < ten_seconds {
+						if exit_flag_requested.load(Ordering::Relaxed) {
```

Comment:
please just use ``exit_flag`` directly instead of making a new clone to the ``Arc``